app/controllers/accounts_controller.rb

```diff
@@ -31,9 +31,28 @@ def create
     redirect_to my_account_path
   end
 
+  def archive
+    unless accessed?
+      flash.now.notice = 'Access denied!.'
+      return render :notification
+    end
+
+    account.update!(archived_at: Time.current)
+    AccountShare.where(account_id: account).delete_all
+    AccountAutomaticTopupConfig.where(to_account: account).delete_all
```

Comment:
let's also archive `AccountAutomaticTopupConfig`

Reply:
@VladislavSokov I have a better idea. Let's not change anything except the account. However, the place that selects the top-up configurations should be changed to filter those related to deleted accounts.